### ISSUE TITLE
[Bug]: Fody + .NET 9 project: Hot Reload (EnC) broken due to wrong/duplicate System.Runtime reference (CS7038)

### ISSUE THREAD

**Issue by jaredthirsk** [reactiveui/ReactiveUI#3937]
### Describe the bug 🐞

Hot reload fails (both dotnet watch, and in VS2022 Preview) on .NET 9.0 project with ReactiveUI.Fody.Helpers.

If I remove [Reactive] attribute and ReactiveUI.Fody (I think version [19.5.41](https://www.nuget.org/packages/ReactiveUI.Fody/19.5.41)), hot reload works.

(If I add [ReactiveUI.SourceGenerators](https://www.nuget.org/packages/ReactiveUI.SourceGenerators/), and use their different [Reactive] attribute on a field, it works.)

### Step to reproduce

1. Create .NET 9 project
2. Add ReactiveUI.Fody nuget 
3. Make a class with [Reactive] property
4. Debug in VS2022 Preview, or `dotnet watch`
5. Try to make a change for hot reload
6. Fails with error:

```
dotnet watch :x: error CS7038: Failed to emit module 'Xyz.Core': Changing the version of an assembly reference is not allowed during debugging: 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' changed version to '9.0.0.0'.
```

7. If I inspect the DLL with dotPeek, I see there are two references for System.Runtime, version 8.0 and 9.0.  I suspect this is where the problem is.  Microsoft's Hot Reload should ideally detect that both were there before and after so no version was changed, but I suspect this scenario of two versions of the same DLL is not supposed to happen (I'm not an expert though, so I can't say for sure.)

### Expected behavior

- Expectation: Hot reload should work
- Expectation: DLL shouldn't have two references to System.Runtime (v8, v9)

### IDE

Visual Studio 2022

### Operating system

Windows 10

### ReactiveUI Version

19.5.41

### Workarounds

1.  If I change my project back to .NET 8, hot reload works.
2. If I drop ReactiveUI.Fody and use ReactiveUI.SourceGenerators instead, it works. (Though it has breaking changes to my code.)

### Additional information ℹ️

Related issues:
- https://github.com/dotnet/roslyn/issues/76284
- https://github.com/Fody/Fody/issues/1044

**Comment by ChrisPulman** [comment]
Have you tried [ReactiveUI.SourceGenerators](https://github.com/reactiveui/ReactiveUI.SourceGenerators) V2.1.8?

**Comment by jaredthirsk** [comment]
> Have you tried [ReactiveUI.SourceGenerators](https://github.com/reactiveui/ReactiveUI.SourceGenerators) V2.1.8?

Yes, it works fine (I am on 2.0.17) and I am replacing ReactiveUI.Fody with it.

**Comment by ChrisPulman** [comment]
I will close this as the recommendation is to use ReactiveUI.SourceGenerators to replace ReactiveUI.Fody, thank you for raising this.
